src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigClientImplTest.java

```diff
@@ -0,0 +1,377 @@
+  }
+
+  @Test
+  public void testGetTemplateErrorWithZeroContentResponse() {
+    for (int code : HTTP_ERRORS) {
+      response.setStatusCode(code).setZeroContent();
+
+      try {
+        client.getTemplate();
+        fail("No error thrown for HTTP error");
+      } catch (FirebaseRemoteConfigException error) {
+        checkExceptionFromHttpResponse(error, HTTP_2_ERROR.get(code), null,
+                "Unexpected HTTP response with status: " + code + "\nnull");
+      }
+      checkGetRequestHeader(interceptor.getLastRequest());
+    }
+  }
+
+  @Test
+  public void testGetTemplateErrorWithMalformedResponse() {
+    for (int code : HTTP_ERRORS) {
+      response.setStatusCode(code).setContent("not json");
+
+      try {
+        client.getTemplate();
+        fail("No error thrown for HTTP error");
+      } catch (FirebaseRemoteConfigException error) {
+        checkExceptionFromHttpResponse(error, HTTP_2_ERROR.get(code), null,
+                "Unexpected HTTP response with status: " + code + "\nnot json");
+      }
+      checkGetRequestHeader(interceptor.getLastRequest());
+    }
+  }
+
+  @Test
+  public void testGetTemplateErrorWithDetails() {
+    for (int code : HTTP_ERRORS) {
+      response.setStatusCode(code).setContent(
+              "{\"error\": {\"status\": \"INVALID_ARGUMENT\", \"message\": \"test error\"}}");
+
+      try {
+        client.getTemplate();
+        fail("No error thrown for HTTP error");
+      } catch (FirebaseRemoteConfigException error) {
+        checkExceptionFromHttpResponse(error, ErrorCode.INVALID_ARGUMENT);
+      }
+      checkGetRequestHeader(interceptor.getLastRequest());
+    }
+  }
+
+  @Test
+  public void testGetTemplateErrorWithCanonicalCode() {
+    for (int code : HTTP_ERRORS) {
+      response.setStatusCode(code).setContent(
+              "{\"error\": {\"status\": \"NOT_FOUND\", \"message\": \"test error\"}}");
+
+      try {
+        client.getTemplate();
+        fail("No error thrown for HTTP error");
+      } catch (FirebaseRemoteConfigException error) {
+        checkExceptionFromHttpResponse(error, ErrorCode.NOT_FOUND);
+      }
+      checkGetRequestHeader(interceptor.getLastRequest());
+    }
+  }
+
+  @Test
+  public void testGetTemplateErrorWithRcError() {
+    for (int code : HTTP_ERRORS) {
+      response.setStatusCode(code).setContent(
+              "{\"error\": {\"status\": \"INVALID_ARGUMENT\", "
+                      + "\"message\": \"[INVALID_ARGUMENT]: test error\"}}");
+
+      try {
+        client.getTemplate();
+        fail("No error thrown for HTTP error");
+      } catch (FirebaseRemoteConfigException error) {
+        checkExceptionFromHttpResponse(error, ErrorCode.INVALID_ARGUMENT,
+                RemoteConfigErrorCode.INVALID_ARGUMENT, "[INVALID_ARGUMENT]: test error");
+      }
+      checkGetRequestHeader(interceptor.getLastRequest());
+    }
+  }
+
+  @Test
+  public void testGetTemplateErrorWithDetailsAndNoCode() {
```

Comment:
Duplicate of `testGetTemplateErrorWithDetails`

Reply:
Good catch! removed both `testGetTemplateErrorWithCanonicalCode` and `testGetTemplateErrorWithDetailsAndNoCode`